crates/trigger/src/cli.rs

```diff
@@ -114,21 +117,43 @@ where
         let working_dir = std::env::var(SPIN_WORKING_DIR).context(SPIN_WORKING_DIR)?;
         let locked_url = std::env::var(SPIN_LOCKED_URL).context(SPIN_LOCKED_URL)?;
 
-        let loader = TriggerLoader::new(working_dir, self.allow_transient_write);
+        // TODO: I assume there is a way to do this with a single let mut loader: Box<dyn Loader>
+        // variable instead of the entire executor.
+        let executor: Executor = match self.oci {
+            true => {
+                let loader = OciTriggerLoader::new(working_dir, self.allow_transient_write).await?;
```

Comment:
Do the triggers really need to know about OCI?  One of the lovely things about the lockfile design was that it got the application to a place where _triggers didn't need to care_ - `spin up` just handed over the lockfile URL and the trigger took it from there.

(Maybe I've been avoiding some horrible truth about lockfiles but that was the way I understood it anyway!)

Reply:
It needs to know whether it is running an application from a relative file path or from the global cache.